src/sync/sync_manager.hpp

```diff
@@ -66,7 +66,8 @@ friend class SyncSession;
     // Configure the metadata and file management subsystems. This MUST be called upon startup.
     void configure_file_system(const std::string& base_file_path,
                                MetadataMode metadata_mode=MetadataMode::Encryption,
-                               util::Optional<std::vector<char>> custom_encryption_key=none);
+                               util::Optional<std::vector<char>> custom_encryption_key=none,
+                               bool reset_metadata_on_error=false);
```

Comment:
The only case that comes to mind is if binding specifies custom encryption key and needs to check if it's valid to try another one. My point is while binding provides encryption key it should be responsible for reseting metadata. Do you think `reset_metadata_on_error` should be `true` by default or should just be removed?